android/app/src/main/java/org/fossasia/openevent/viewmodels/SponsorsFragmentViewModel.java

```diff
@@ -0,0 +1,38 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+
+import org.fossasia.openevent.data.Sponsor;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+
+import java.util.List;
+
+import io.realm.RealmResults;
+
+public class SponsorsFragmentViewModel extends ViewModel {
+
+    private MutableLiveData<List<Sponsor>> sponsorsList;
+    private RealmDataRepository realmRepo;
+    private RealmResults<Sponsor> realmResults;
+
+    public SponsorsFragmentViewModel() {
+        realmRepo = RealmDataRepository.getDefaultInstance();
+    }
+
+    public LiveData<List<Sponsor>> getSponsors() {
+        sponsorsList = new MutableLiveData<>();
```

Comment:
Actually this is not the right approach. It creates a new LiveData object and changeListener everytime the method is called. Create a new LiveData instance and attach changeListener only when the LiveData instance is null.

Reply:
Okay sounds good